### PR TITLE
Input Manager

### PR DESCRIPTION
**Pull request by LucaHaverty**
### Description
An input management system that can control Synthesis using both keyboard and controller with customizable. Some controls are global and some are robot specific. The input manager works with new behavior scripts to move robots.

### Objectives
- [x] Input management system
- [x] Ability to change inputs from the controls modal
- [x] Arcade drive behavior ([JIRA Issue](https://jira.autodesk.com/browse/AARD-1663))
- [x] Generic arm behavior ([JIRA Issue](https://jira.autodesk.com/browse/AARD-1664))
- [x] Generic elevator behavior ([JIRA Issue](https://jira.autodesk.com/browse/AARD-1674))

**(the wacky slider joint behavior is fixed in [Elevator Bug](https://github.com/Autodesk/synthesis/pull/983))**

[JIRA Issue](https://jira.autodesk.com/browse/AARD-1658)

### CODE REVIEW

**Review by HunterBarclay — CHANGES_REQUESTED** [review]
image

The joint inputs are appearing under the global section.

**Review by PepperLola — DISMISSED** [review]
It works pretty well (except for the slider I think). I think something about the keybind detection needs to change to work on Mac though. When doing <kbd>Shift + 1</kbd> it instead registers as <kbd>Shift + !</kbd>, and with alt it becomes <kbd>Alt + ¡</kbd> rather than detecting the base key (1) and the modifier (Shift). I can rebind the controls to <kbd>Alt + ¡</kbd> and it'll work properly, although it seems that joint 2 on the default TMM 2018 robot doesn't do anything and joint 3 controls both stages of the elevator.

It also looks like either the joint inputs aren't updating (maybe since they're being categorized as global?) or they're being added to robots that shouldn't have them, because when only Dozer is loaded I still see the arm and slider joints from the TMM 2018 bot.

**Review by a-crowell — APPROVED** [review]
Other than the multibot issues (controlling all robots simultaneously and controls from other robots showing up) which sounds like it will be solved later--looks great!

Someday:
* The stages of the 2471 elevator could have less impact on each other (they can move each other almost min to max), but the elevator fix commit was still a huge improvement and seems good enough for now.
* Shift/alt/ctrl combos don't do anything. Easily fixed by not using them, so again, future problem.



Looking over this was super impressive. Great job!

**Review by HunterBarclay — CHANGES_REQUESTED** [review]
Modifier keys still don't seem to work. `Shift + 1` originally worked but after rebinding, no modifier keys responded (with or without)

**Comment by LucaHaverty** [comment]
> Modifier keys still don't seem to work. `Shift + 1` originally worked but after rebinding, no modifier keys responded (with or without)

Could you give me exact steps to recreate? I haven't been able to get modifier keys to not work, but Ayla seems to have found some issues with them too.

File: fission/src/systems/simulation/stimulus/WheelStimulus.ts [hunk 1]
```diff
@@ -34,9 +34,9 @@ class WheelRotationStimulus extends EncoderStimulus {
         this._wheel = wheel;
     }
 
-    public Update(deltaT: number): void {
+    public Update(_: number): void {
         if (this._accum) {
-            this._wheelRotationAccum += this._wheel.GetAngularVelocity() * deltaT;
+            this._wheelRotationAccum += 0;
```
Comment (HunterBarclay) [inline fission/src/systems/simulation/stimulus/WheelStimulus.ts]: Why'd you remove this?

Comment (LucaHaverty) [reply fission/src/systems/simulation/stimulus/WheelStimulus.ts]: I wasn't sure what it did and removed it while debugging wheel behavior, but I just now added it back in my most recent commit.

Comment (HunterBarclay) [reply fission/src/systems/simulation/stimulus/WheelStimulus.ts]: @LucaHaverty K, looks like when you added it back, you accidentally pasted in the entire line again. Fixed it just now.

**Review by HunterBarclay — APPROVED** [review]
Looks great. Found a lot of little bugs with this one, so that's always good too.

**Pull request merged**